gixy/directives/directive.py

```diff
@@ -68,6 +68,59 @@ def __str__(self):
         return f"{self.name} {' '.join(self.args)};"
 
 
+class MapDirective(Directive):
+    """
+    map $source $destination {
+        default value;
+        key     value;
+    }
+    """
+
+    nginx_name = "map" # XXX: Should also work for "charset_map"
+    provide_variables = True
```

Comment:
Is this necessarily correct? It does provide variables in a sense, but it actually provides the variable `$source` with the respective `$destination` depending on the value.

Reply:
No, it's not correct. Fixing.